src/pages/create/Activity.tsx

```diff
@@ -0,0 +1,282 @@
+
+  const { client } = useDeskproAppClient();
+
+  const { object, parentId } = useParams();
+
+  const navigate = useNavigate();
+
+  const [schema, setSchema] = useState<ZodObject<any>>(z.object({}));
+
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    setValue,
+    watch,
+  } = useForm<any>({
+    resolver: zodResolver(schema as ZodObject<any>),
+  });
+
+  const type = watch("Type");
+
+  useEffect(() => {
+    if (parentId) {
+      setValue(object as string, parentId);
+    }
+  }, [parentId, setValue, object]);
+
+  const people = useQueryWithClient(
+    [QueryKey.ACCOUNT_BY_ID],
+    (client) =>
+      getObjectsByQuery(
+        client,
+        "SELECT Id, Name FROM User WHERE UserPermissionsSFContentUser=true",
+        200
+      ),
+    {
+      enabled: !!activityTypes
+        .find((e) => e.value === type)
+        ?.fields.findIndex((e) => e.name === "OwnerId"),
+    }
+  );
+
+  const ActivityMetadata = useQueryWithClient(
+    [QueryKey.ACTIVITY_METADATA, type],
+    (client) =>
+      getObjectMeta(
+        client,
+        activityTypes.find((e) => e.label === type)?.value as string
+      ),
+    {
+      enabled: !!type,
+    }
+  );
+
+  useEffect(() => {
+    if (!type || !ActivityMetadata.data) return;
+
+    const activityType = activityTypes.find((e) => e.label === type);
+
+    setSchema(
+      getActivitySchema(
+        activityType?.fields as Field[],
+        activityType?.value as string
+      )
+    );
+    setValue(`${activityType?.value}Subtype`, activityType?.label);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [ActivityMetadata.data, watch, type]);
+
+  const submit = async (data: any) => {
+    if (!client) return;
+
+    setIsSubmitting(true);
+
+    delete data.Type;
+
+    if (data.EventSubtype) {
+      data.DurationInMinutes = Math.abs(
+        (new Date(data.EndDateTime as string).getTime() -
+          new Date(data.ActivityDateTime as string).getTime()) /
+          1000 /
+          60
+      );
+    }
+
+    await postData(
+      client,
+      activityTypes.find((e) => e.label === type)?.value as string,
+      data
+    );
+
+    navigate(-1);
+  };
+
+  return (
+    <form onSubmit={handleSubmit(submit)} style={{ margin: "5px" }}>
+      <DropdownSelect
+        title="Type"
+        value={type || ""}
+        onChange={(e) => {
+          setValue("Type", e);
+        }}
+        error={!!errors.Type}
+        data={activityTypes}
+        keyName={"label"}
+        valueName={"label"}
+      />
+      {type && (
+        <Stack vertical gap={12}>
+          {activityTypes
+            .find((e) => e.label === type)
+            ?.fields.map((field, i) => {
+              let values: { value: string; key: string }[] = [];
+
+              if (field.label === "Type") return;
+
+              if (field.type === "dropdown") {
+                switch (field.name) {
+                  case "Status":
```

Comment:
Why is `Status` and `OwnerID` hard-coded? If these are associations/lookups, then we need to build the field from salesforce metadata alone.

A salesforce user may add/remove fields and associations at any time, so they're unpredictable... granted `OwnerID` _probably_ isn't, but I think it should be handled like any other field.